pkg/infrastructure/kafka.go

```diff
@@ -101,3 +103,75 @@ func GetKafkaServiceURI(cli *client.Client, infra *v1alpha1.KogitoInfra) (uri st
 	}
 	return
 }
+
+// GetReadyKafkaInstanceName fetches for the Kafka Instance linked with the given Kogito and checks if it is ready
+func GetReadyKafkaInstanceName(cli *client.Client, infra *v1alpha1.KogitoInfra) (kafka string, err error) {
+	kafka = ""
+	err = nil
+	if &infra == nil || &infra.Status == nil || &infra.Status.Kafka == nil || len(infra.Status.Kafka.Name) == 0 {
+		return
+	}
+
+	instance, err := GetKafkaInstanceWithName(infra.Status.Kafka.Name, infra.Namespace, cli)
+	if err != nil {
+		return
+	}
+	if instance != nil {
+		uri := ResolveKafkaServerURI(instance)
+		if len(uri) > 0 {
+			kafka = instance.Name
+		}
+	}
+	return
+}
+
+// ResolveKafkaServerURI returns the uri of the kafka instance
+func ResolveKafkaServerURI(kafka *v1beta1.Kafka) string {
+	if kafka != nil {
+		if len(kafka.Status.Listeners) > 0 {
+			for _, listenerStatus := range kafka.Status.Listeners {
+				if listenerStatus.Type == "plain" && len(listenerStatus.Addresses) > 0 {
+					for _, listenerAddress := range listenerStatus.Addresses {
+						if len(listenerAddress.Host) > 0 && listenerAddress.Port > 0 {
+							return fmt.Sprintf("%s:%d", listenerAddress.Host, listenerAddress.Port)
+						}
+					}
+				}
+			}
+		}
+	}
+	return ""
+}
+
+// ResolveKafkaServerReplicas returns the number of replicas of the kafka instance
+func ResolveKafkaServerReplicas(kafka *v1beta1.Kafka) int32 {
+	if kafka != nil {
+		if kafka.Spec.Kafka.Replicas > 0 {
+			return kafka.Spec.Kafka.Replicas
+		}
+		return 1
```

Comment:
Why the 1 is returned here when 'kafka.Spec.Kafka.Replicas' is 0 or less?

Reply:
Because the code sets the default value of this property is 0 if no other value is provided, and a value which is less than or equal to 0 doesn't make any sense. So set the replicas to 1 in this case.